pkg/k8s/auth.go

```diff
@@ -0,0 +1,29 @@
+package k8s
+
+import "fmt"
+
+const (
+	GKE_GCLOUD_AUTH_PLUGIN_MISSING      = "no Auth Provider found for name \"gcp\""
+	GKE_GCLOUD_AUTH_PLUGIN_INSTALL_HINT = `
+Hint:
+  * Install gke-gcloud-auth-plugin by following https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
+`
+
+	EKS_AUTH_PLUGIN_OUTDATED     = "exec plugin: invalid apiVersion \"client.authentication.k8s.io/v1alpha1\""
+	EKS_AUTH_PLUGIN_UPGRADE_HINT = `
+Hint:
+  * Upgrade AWS CLI by following https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+  * Update your kubeconfig by following https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html
+`
+)
+
+func wrapHintIfAuthError(err error) error {
+	switch err.Error() {
+	case EKS_AUTH_PLUGIN_OUTDATED:
+		return fmt.Errorf("%v\n%s", err, EKS_AUTH_PLUGIN_UPGRADE_HINT)
```

Comment:
its preferable to try return an error that doesn;t assumes its printed (\r\n) but I dont now how the catcher that prints the error work, can you try to see if it makes sense so this funciton can return an error without assuming its printed ( or either print it here)

Reply:
output example:

> 
WARN[0007] no Auth Provider found for name "gcp"        
Hint:
\* Install gke-gcloud-auth-plugin by following https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke
ERRO[0015] authentication failure to gke_strange-mason-318808_us-central1-b_dev-cluster-c453

Reply:
is it a warning or an error ? can the system be installed if this occurs?